File: analogagenda.client/src/app/DTOs/idea.dto.ts
export interface IdeaSessionSummaryDto {
  id: string;
  displayLabel: string;
}

export interface IdeaDto {
  id: string;
  title: string;
  description: string;
  outcome?: string;
  connectedSessionIds?: string[];
  connectedSessions?: IdeaSessionSummaryDto[];
}

File: Database/DTOs/IdeaDto.cs
namespace Database.DTOs;

public class IdeaSessionSummaryDto
{
    public string Id { get; set; } = string.Empty;
    public string DisplayLabel { get; set; } = string.Empty;
}

public class IdeaDto
{
    public string Id { get; set; } = string.Empty;

    public string Title { get; set; } = string.Empty;

    public string Description { get; set; } = string.Empty;

    public string Outcome { get; set; } = string.Empty;

    public List<string> ConnectedSessionIds { get; set; } = [];

    public List<IdeaSessionSummaryDto> ConnectedSessions { get; set; } = [];
}
